### PR TITLE
arch specific requs

### PR DESCRIPTION
**Pull request by tarekziade**
Cleanup and tweak the requirements since we have different libs on aarch64

### CODE REVIEW

**Comment by tarekziade** [comment]
> While change looks good, it sounds like a potentially serious problem that we use different versions of libraries per platform, should we pick libraries that are equaly well supported on both architectures?

I think it's best to support each architecture's latest version and document the gap. In an ideal world, we close the gap in the aarch64 ones. It's probable that we can rebuild them at install time from the source

File: Makefile [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
 	$(PYTHON) -m venv .
 	bin/pip install --upgrade pip
 
 install: bin/python
-	bin/pip install -r requirements.txt
+	bin/pip install -r requirements/$(ARCH).txt
 
 dev: install
-	bin/pip install -r test-requirements.txt
+	bin/pip install -r requirements/tests.txt
```
Comment (wangch079) [inline Makefile:14]: I wonder what is this used for?

File: Makefile [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
 bin/elastic-ingest: bin/python
-	bin/pip install -r requirements.txt
+	bin/pip install -r requirements/$(ARCH).txt
 	bin/python setup.py develop
 
 bin/black: bin/python
-	bin/pip install -r requirements.txt
-	bin/pip install -r test-requirements.txt
+	bin/pip install -r requirements/$(ARCH).txt
+	bin/pip install -r test-requirements/$(ARCH).txt
```
Comment (wangch079) [inline Makefile:22]:
```suggestion
	bin/pip install -r requirements/tests.txt
```

File: Makefile [hunk 3]
```diff
@@ -1,29 +1,30 @@
 .PHONY: test lint run ftest install dev
 
 PYTHON=python3.10
+ARCH=$(shell uname -m)
```
Comment (wangch079) [inline Makefile:4]: Should we have a fallback to set `ARCH` to `framework` if it's not one of the supported values?

File: Makefile (hunk 1 above)
Comment (tarekziade) [reply Makefile:14]: you mean the tests requirement? it's all the packages use to run tests, that are not installed in production to *run* the service

File: Makefile (hunk 3 above)
Comment (tarekziade) [reply Makefile:4]: good idea we can fallback to `x86_64` (also in setup.py)

Comment (tarekziade) [reply Makefile:4]: actually just do `make install ARCH=x86_64`

**Comment by tarekziade** [comment]
buildkite test this

**Comment by j-bennet** [comment]
I wonder if you could keep the same `requirements.txt`, but use the environment markers:

https://peps.python.org/pep-0496/

**Comment by tarekziade** [comment]
> I wonder if you could keep the same `requirements.txt`, but use the environment markers:
> 
> https://peps.python.org/pep-0496/

great idea, let's do it as a follow up

**Review by j-bennet — APPROVED** [review]
👍

**Pull request merged**